src/logic/basic.lean

```diff
@@ -1087,6 +1089,9 @@ theorem forall_prop_of_true {p : Prop} {q : p → Prop} (h : p) : (∀ h' : p, q
 theorem exists_prop_of_true {p : Prop} {q : p → Prop} (h : p) : (∃ h' : p, q h') ↔ q h :=
 @exists_const (q h) p ⟨h⟩
 
+lemma exists_iff_of_forall {p : Prop} {q : p → Prop} (h : ∀ h, q h) : (∃ h, q h) ↔ p :=
```

Comment:
Is this a special case of the previous?

Reply:
How so? The previous has ` q h` on the RHS, not `p`.

Reply:
The LHS matches though, and it looks like the RHS should be an easy lemma away? It's fine as is though, your proof is short enough for sure

Reply:
I'm not so sure, because the assumptions are different.